src/taipy/config/data_node/data_node_config.py

```diff
@@ -76,34 +84,52 @@ class DataNodeConfig:
     # Pickle
     _OPTIONAL_DEFAULT_PATH_PICKLE_PROPERTY = "default_path"
     _OPTIONAL_DEFAULT_DATA_PICKLE_PROPERTY = "default_data"
-
-    _REQUIRED_PROPERTIES: Dict[str, List] = {_STORAGE_TYPE_VALUE_PICKLE: [],
-                            _STORAGE_TYPE_VALUE_SQL: [_REQUIRED_DB_USERNAME_SQL_PROPERTY,
-                                                      _REQUIRED_DB_PASSWORD_SQL_PROPERTY,
-                                                      _REQUIRED_DB_NAME_SQL_PROPERTY,
-                                                      _REQUIRED_DB_ENGINE_SQL_PROPERTY,
-                                                      _REQUIRED_READ_QUERY_SQL_PROPERTY,
-                                                      _REQUIRED_WRITE_TABLE_SQL_PROPERTY ],
-                            _STORAGE_TYPE_VALUE_CSV: [],
-                            _STORAGE_TYPE_VALUE_EXCEL: [],
-                            _STORAGE_TYPE_VALUE_IN_MEMORY: [],
-                            _STORAGE_TYPE_VALUE_GENERIC: [_REQUIRED_READ_FUNCTION_GENERIC_PROPERTY,
-                                                          _REQUIRED_WRITE_FUNCTION_GENERIC_PROPERTY]}
-
-    _OPTIONAL_PROPERTIES = {_STORAGE_TYPE_VALUE_GENERIC: [_OPTIONAL_READ_FUNCTION_PARAMS_GENERIC_PROPERTY,
-                                                          _OPTIONAL_WRITE_FUNCTION_PARAMS_GENERIC_PROPERTY],
-                            _STORAGE_TYPE_VALUE_CSV: [_OPTIONAL_EXPOSED_TYPE_CSV_PROPERTY,
-                                                      _OPTIONAL_DEFAULT_PATH_CSV_PROPERTY,
-                                                      _OPTIONAL_HAS_HEADER_CSV_PROPERTY],
-                            _STORAGE_TYPE_VALUE_EXCEL: [_OPTIONAL_EXPOSED_TYPE_EXCEL_PROPERTY,
-                                                        _OPTIONAL_DEFAULT_PATH_EXCEL_PROPERTY,
-                                                        _OPTIONAL_HAS_HEADER_EXCEL_PROPERTY,
-                                                        _OPTIONAL_SHEET_NAME_EXCEL_PROPERTY],
-                            _STORAGE_TYPE_VALUE_IN_MEMORY: [_OPTIONAL_DEFAULT_DATA_IN_MEMORY_PROPERTY],
-                            _STORAGE_TYPE_VALUE_SQL: [_OPTIONAL_EXPOSED_TYPE_SQL_PROPERTY],
-                            _STORAGE_TYPE_VALUE_PICKLE: [_OPTIONAL_DEFAULT_PATH_PICKLE_PROPERTY,
-                                                         _OPTIONAL_DEFAULT_DATA_PICKLE_PROPERTY],
-                            }
+    # JSON
+    _REQUIRED_DEFAULT_PATH_JSON_PROPERTY = "default_path"
```

Comment:
What about the Data class representing the json object?

Reply:
So the loaded json data will be represented as a dictionary right? Imo, we can also allow the user to provide their custom data class like JR mentioned, data class is easy to use and manipulate, if the data class is not provided, we can keep the default dictionary behavior. What do you think?

Reply:
Yes, that's what I meant.